judge/views/contests.py

```diff
@@ -62,20 +63,23 @@ def get_queryset(self):
         return Contest.get_visible_contests(self.request.user)
 
 
-class ContestList(DiggPaginatorMixin, TitleMixin, ContestListMixin, ListView):
+class ContestList(QueryStringSortMixin, DiggPaginatorMixin, TitleMixin, ContestListMixin, ListView):
     model = Contest
     paginate_by = 20
     template_name = 'contest/list.html'
     title = gettext_lazy('Contests')
     context_object_name = 'past_contests'
+    all_sorts = frozenset(('name', 'user_count', 'start_time'))
+    default_desc = frozenset(('name', 'user_count'))
+    default_sort = '-start_time'
 
     @cached_property
     def _now(self):
         return timezone.now()
 
     def _get_queryset(self):
         return super(ContestList, self).get_queryset() \
-            .order_by('-start_time', 'key').prefetch_related('tags', 'organizations', 'organizers')
+            .order_by(self.order).prefetch_related('tags', 'organizations', 'organizers')
```

Comment:
Currently, active and future contests are not sorted here: see [here](https://github.com/DMOJ/online-judge/blob/master/judge/views/contests.py#L100). Thus, only past and current contests are affected. This can be confusing, since the order the tables are displayed are active, current, future, past (alternating between affected by sorting and not affected).

I think we should only make past contests affected by this sorting, rather than past and current contests.

Reply:
I have added the secondary order back. Do you think it would be a good idea to instead make future contests also sortable? The order would then be active (not sorted), current (sorted), future (sorted), and past (sorted).

Reply:
I think sorting future contests would be confusing, since by default, it is sorted by `start_time` (as opposed to `-start_time`). This means that when you click on that column to sort, the sort order would not change for the future contests table (until you click again). Another issue is we would now have three states: sorted by `start_time`, sorted by `-start_time`, and default order.

Reply:
There is only `start_time` and the default order now, similar to `user.py` and `problem.py`. Also, current contests are no longer affected by the sorting.